internal/rbd/rbd_util.go

```diff
@@ -426,9 +426,15 @@ func flattenClonedRbdImages(ctx context.Context, snaps []snapshotInfo, pool, mon
 		util.ErrorLog(ctx, "failed to open connection %s; err %v", rv, err)
 		return err
 	}
-	for _, s := range snaps {
-		if s.Namespace.Type == "trash" {
-			rv.RbdImageName = s.Namespace.OriginalName
+
+	for _, snapInfo := range snaps {
+		// get original snap name for the snapshots in trash namespace
+		origSnapName, err := rv.getOrigSnapName(snapInfo.Id)
```

Comment:
why do we need to getOrigSnapName for all snapshots?

Reply:
Inside the function, we are doing for Trash snaps only.